### PR TITLE
Add set_re*_headers

### PR DESCRIPTION
**Pull request by alex-schneider**

### CODE REVIEW

File: handler/proxy.go [hunk 1]
```diff
@@ -379,18 +379,20 @@ func (p *Proxy) SetRoundtripContext(req *http.Request, beresp *http.Response) {
...
 		}
 	}
 
-	for _, ctxBody := range p.options.Context {
-		options, err := NewCtxOptions(attrCtx, evalCtx, ctxBody)
-		if err != nil {
-			p.log.WithField("parse config", p.String()).Error(err)
+	for _, ctx := range attrCtx {
+		for _, ctxBody := range p.options.Context {
+			options, err := NewCtxOptions(ctx, evalCtx, ctxBody)
```
Comment (malud) [inline handler/proxy.go:390]: We should add a deprecated log if `options` is not empty for related `ctx`. This should be a warning with a hint which option Couper prefers now.

**Review by malud — CHANGES_REQUESTED** [review]
lgtm, except the missing log

File: handler/proxy.go (hunk 1 above)
Comment (filex) [reply handler/proxy.go:390]: can we do that during parsing/startup? on every single request, that would be too much.

Comment (alex-schneider) [reply handler/proxy.go:390]: No, we cannot. First, it's a temporary solution until we stop supporting `re*_headers`. Secondly, we would lose the flexibility of the request context ala `x-foo = req.xxx`

Comment (malud) [reply handler/proxy.go:390]: Hm why we lose anything from this feature with a log? This is indeed possible on startup, checking the backend attributes for the outdated names should do the trick.

**Review by malud — APPROVED** [review]

**Pull request merged**